datachannel.go

```diff
@@ -175,15 +175,8 @@ func (c *DataChannel) Read(p []byte) (int, error) {
 func (c *DataChannel) ReadDataChannel(p []byte) (int, bool, error) {
 	for {
 		n, ppi, err := c.stream.ReadSCTP(p)
-		if err == io.EOF {
```

Comment:
Are we required to lose this? It's defined in the data channel spec:
https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-6.7

>  Closing of a data channel MUST be signaled by resetting the
   corresponding outgoing streams [RFC6525].  This means that if one
   side decides to close the data channel, it resets the corresponding
   outgoing stream.  When the peer sees that an incoming stream was
   reset, it also resets its corresponding outgoing stream.  Once this
   is completed, the data channel is closed.

Reply:
Hmmm.  I may have read it wrong.
```
When the peer sees that an incoming stream was reset, it also resets its corresponding outgoing stream. 
```
In this sentence, "the peer" is referred to SCTP entity or application?

In TCP, when a node receives FIN, it goes into CLOSE_WAIT state and application can still send data: https://upload.wikimedia.org/wikipedia/commons/f/f6/Tcp_state_diagram_fixed_new.svg

If we did not make this change, TestEOF would fail.

Reply:
Actually, with non-detach case, if a node receives EOF on a datachannel, OnClose callback is made.  I guess it is a bit weird to have the channel actually half-open after OnClose is made...  How about enabling this half-open (CLOSE_WAIT) state only with detached datachannel?

Reply:
FYI: https://tools.ietf.org/html/rfc6458#section-4.1.7

Reply:
rfc6458 does not seem to assume rfc6525 (reconfig) because it is true to SCTP (rfc4960 alone) only has Shutdown or Abort of association. rfc6525 is pretty close to allow TCP socket's SHUT_WR operation, but again the above sentence is still a question...

Reply:
Regarding half-open (CLOSE_WAIT) state: a data channel is defined as bidirectional: https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-6.4

> Data channels are defined such that their accompanying application-
   level API can closely mirror the API for WebSockets, which implies
   bidirectional streams of data and a textual field called 'label' used
   to identify the meaning of the data channel.

Same on the protocol level: https://tools.ietf.org/html/draft-ietf-rtcweb-data-protocol-09#section-4

> The Data Channel Establishment Protocol is a simple, low-overhead way
   to establish bidirectional Data Channels over an SCTP association
   with a consistent set of properties.

Therefore, It seems unlikely that this use-case would come up.

Reply:
Re-opened. @enobufs let us know if you agree.